### PR TITLE
Bank details pattern

### PR DESCRIPTION
**Pull request by adamsilver**

### CODE REVIEW

File: src/patterns/bank-details/index.md.njk [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+Some users with older bank or building society accounts might have account numbers that are 9 or 10 digits long. Research is needed to confirm that these users know how to shorten their account number to fit into the field.
+
+Research is needed on how to ask users for their bank details in the context of setting up a Direct Debit.
+
+More research is needed on whether bank details should be shown or not on a ‘Check your answers’ page, and how that affects this pattern.
+
+More research is also needed on the best way to handle international bank accounts so that users are only asked for information that’s needed based on the location of their bank account.
+
+This pattern doesn’t currently allow users to provide details of a credit union account. Research is needed on the information required to make payments to credit union accounts and how best to ask credit union members for that information.
```
Comment (m-green) [inline src/patterns/bank-details/index.md.njk]:
```suggestion
This pattern doesn’t currently allow users to provide details of a credit union account. Research is needed on what information services need to make payments to credit union accounts, and how to ask for it.
```

File: src/patterns/bank-details/index.md.njk [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+{% from "_example.njk" import example %}
+
+{{ example({group: "patterns", item: "bank-details", example: "default", html: true, nunjucks: true, open: false, size: "xl"}) }}
+
+## When to use this pattern
+
+Follow this pattern if you need users to provide their bank details so you can pay them.
+
+Only ask for bank details securely within your service. For example, do not ask users to share their bank details by email.
```
[the remark this answers is not in the document]
Comment (amyhupe) [reply src/patterns/bank-details/index.md.njk]: Yep - I'll change that.

File: src/patterns/bank-details/index.md.njk [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+You may also need to ask users for bank details so that they can pay by Direct Debit. This pattern does not yet cover that use case.
```
[the remark this answers is not in the document]
Comment (amyhupe) [reply src/patterns/bank-details/index.md.njk]: Good shout - I've just made a small edit which is to include the word "yet" as the intention is to add this in at some stage and it'd be good to highlight that to the community - hopefully someone can contribute it.

File: src/patterns/bank-details/index.md.njk [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+If your service lets users set up Direct Debit payments, first check if you can use [GOV.UK Pay](https://www.payments.service.gov.uk/direct-debit/).
+
+## How it works
+
+Make it as easy as possible for users to enter their bank details.
```
[the remark this answers is not in the document]
Comment (amyhupe) [reply src/patterns/bank-details/index.md.njk]: Yep happy to lose it.

File: src/patterns/bank-details/index.md.njk [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+Not everyone has a bank account or wants to share their account details online.
+
+Let users choose to pay or get paid by an alternative method if needed.
+
+Adapt this question depending on what payment options your users need and what your service can support.
+
+{{ example({group: "patterns", item: "bank-details", example: "branch", html: true, nunjucks: true, open: false, size: "xl"}) }}
+
+### International bank account details
```
[the remark this answers is not in the document]
Comment (amyhupe) [reply src/patterns/bank-details/index.md.njk]: No, this is great. Thanks Mark.

File: src/patterns/bank-details/index.md.njk [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+Roll numbers are only used by some building societies for certain types of account, so make this field optional.
+
+However, it is required when a user’s account has one. This is why the label says ‘(if you have one)’.
+
+### Allow different formats for the sort code
+
+Let users enter their sort code in whatever format is familiar to them. Allow additional spaces, hyphens and dashes.
+
+Use one input for the sort code. It’s quicker to use and does not need multiple input labels that might be unclear.
```
[the remark this answers is not in the document]
Comment (amyhupe) [reply src/patterns/bank-details/index.md.njk]:
```suggestion
Use one input for the sort code. It allows users to enter their sort code more quickly, and avoids them needing to read and understand multiple input labels.
```

File: src/patterns/bank-details/index.md.njk [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+{{ example({group: "patterns", item: "bank-details", example: "international", html: true, nunjucks: true, open: false, size: "xl"}) }}
+
+### Error messages
+
+Error messages should be styled like this:
+
+{{ example({group: "patterns", item: "bank-details", example: "error", html: true, nunjucks: true, open: false, size: "xl"}) }}
+
+Make sure errors follow the guidance in [error message](https://design-system.service.gov.uk/components/error-message/) and have specific error messages for specific error states.
```
[the remark this answers is not in the document]
Comment (amyhupe) [reply src/patterns/bank-details/index.md.njk]: Unfortunately the capital B is hard-coded at the minute, but in general we use lowercase for pattern names.

File: src/patterns/bank-details/index.md.njk [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+However, it is required when a user’s account has one. This is why the label says ‘(if you have one)’.
+
+### Allow different formats for the sort code
+
+Let users enter their sort code in whatever format is familiar to them. Allow additional spaces, hyphens and dashes.
+
+Use one input for the sort code. It’s quicker to use and does not need multiple input labels that might be unclear.
+
+### Use a branching question if users can choose how to pay or get paid
```
[the remark this answers is not in the document]
Comment (amyhupe) [reply src/patterns/bank-details/index.md.njk]: No that makes sense - good spot - will change.

File: src/patterns/bank-details/index.md.njk [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+The team who developed this pattern only found one service, Claim power of attorney refund, that warns the user that the details will be hidden for their security.
+
+But the GoCardless Direct Debit provider that GOV.UK Pay uses says to display the customer’s bank details back to them before submission.
+
+Some users with older bank or building society accounts might have account numbers that are 9 or 10 digits long. Research is needed to confirm that these users know how to shorten their account number to fit into the field.
+
+Research is needed on how to ask users for their bank details in the context of setting up a Direct Debit.
+
+More research is needed on whether bank details should be shown or not on a ‘Check your answers’ page, and how that affects this pattern.
```
[the remark this answers is not in the document]
Comment (amyhupe) [reply src/patterns/bank-details/index.md.njk]: Yep - I will combine those.

**Pull request merged**